plugins/scanners/scanoss/src/main/kotlin/ScanOss.kt

```diff
@@ -102,26 +95,25 @@ class ScanOss internal constructor(
                 }
         }
 
-        val response = runBlocking {
-            val wfpBody = wfpString.toRequestBody("application/octet-stream".toMediaType())
-            val wfpFile = MultipartBody.Part.createFormData(ARG_FIELD_NAME, FAKE_WFP_FILE_NAME, wfpBody)
-
-            service.scan(wfpFile)
-        }
+        val result = service.scan(
+            wfpString,
+            context.labels["scanOssContext"],
+            context.labels["scanOssId"]?.toIntOrNull() ?: Thread.currentThread().id.toInt()
```

Comment:
Can you provide some information what the context and id parameters are used for? The function docs don't help much:
https://github.com/scanoss/scanoss.java/blob/8598c47987bb92490755972e845bd9fe2a6955d8/src/main/java/com/scanoss/rest/ScanApi.java#L130-L131

That could help to understand why you introduced labels to set them instead of using scanner options. The new labels should also be documented somewhere.

Also, does it really make sense to use the arbitrary thread id as fallback here, instead of just a zero?

Reply:
> Can you provide some information what the context and id parameters are used for?

Unfortunately not, I also do not know more than the function docs and the implementation. Maybe @scanossjeronimo can help here?

> That could help to understand why you introduced labels to set them instead of using scanner options.

TBH, simply because that was the first thing that came to my mind.

> Also, does it really make sense to use the arbitrary thread id as fallback here, instead of just a zero?

Well, I just used the thread ID because that's what the function docs that you quoted say to do... but I agree that such an ID does not seem to be very useful, because it does not outlive the ORT scan run.

In the end, I'm also fine with just hard-coding these parameters to `null` and `0` for now.

Reply:
I'll leave that up to you, I think we need to find out what this conntext and id are to decide if it even makes sense to make them configurable, but we can do that later.